### PR TITLE
Various cleanups

### PR DESCRIPTION
**Pull request by citizen428**
## What type of PR is this? (check all applicable)

- [X] Refactor

## Description

This PR is a hodgepodge of sometimes long overdue cleanups. Since they are all quite distinctive I'll leave inline comments.

## QA Instructions, Screenshots, Recordings

Nothing specific.

### UI accessibility concerns?

None

## Added/updated tests?

- [X] No, and this is why: nothing new here, just some overdue cleanup

## [Forem core team only] How will this change be communicated?

- [X] This change does not need to be communicated, and this is why not: nothing new here, just some overdue cleanup

### CODE REVIEW

File: app/models/organization_membership.rb [hunk 1]
```diff
@@ -1,7 +1,4 @@
 class OrganizationMembership < ApplicationRecord
-  # TODO: [@rhymes]  remove this column, it is unused
```
Comment (citizen428) [inline app/models/organization_membership.rb:2]: Verified via Blazer that this column is unused, so we're dropping it.

File: app/models/profile.rb [hunk 2]
```diff
@@ -17,12 +17,6 @@ def self.refresh_attributes!
     return unless Database.table_available?("profiles")
 
     ProfileField.find_each do |field|
-      # Don't generate accessors for static fields stored on the table.
```
Comment (citizen428) [inline app/models/profile.rb:20]: The corresponding fields have been removed, so we can delete this comment and the guard clause.

File: app/models/settings/general.rb [hunk 3]
```diff
@@ -2,8 +2,6 @@ module Settings
   class General < Base
     self.table_name = "site_configs"
 
-    HEX_COLOR_REGEX = /\A#(\h{6}|\h{3})\z/
```
Comment (citizen428) [inline app/models/settings/general.rb:5]: This constant is no longer used in this file, it was a leftover from the settings refactoring.

File: app/models/user.rb [hunk 4]
```diff
@@ -313,9 +313,7 @@ def cached_followed_tag_names
   end
 
   def suspended?
-    # TODO: [@jacobherrington] After all of our Forems have been successfully deployed,
```
Comment (citizen428) [inline app/models/user.rb:316]: Verified via Blazer that the old "banned" role indeed no longer exists, so we can safely remove this code (which was meant to be extremely temporary).

File: app/models/user.rb [hunk 5]
```diff
@@ -354,10 +352,7 @@ def moderator_for_tags
   end
 
   def comment_suspended?
-    # TODO: [@jacobherrington] After all of our Forems have been successfully deployed,
```
Comment (citizen428) [inline app/models/user.rb:357]: Same as above.

File: app/helpers/application_helper.rb [hunk 6]
```diff
@@ -290,8 +290,6 @@ def admin_config_description(content)
   end
 
   def role_display_name(role)
-    # TODO: [@jacobherrington] After all Forems have successfully deployed and the banned role
```
Comment (citizen428) [inline app/helpers/application_helper.rb:293]: This role has been removed quite a while ago, the helper can now be simplified again.

File: app/queries/admin/moderators_query.rb [hunk 7]
```diff
@@ -4,9 +4,7 @@ class ModeratorsQuery
       state: :trusted
     }.with_indifferent_access.freeze
 
-    # TODO: [@jacobherrington] Once all Forems have been deployed and the data update scripts have
```
Comment (citizen428) [inline app/queries/admin/moderators_query.rb:7]: The roles have been removed, so we can also delete them from the array and remove this comment.

File: app/controllers/async_info_controller.rb [hunk 8]
```diff
@@ -25,16 +24,6 @@ def base_data
     end
   end
 
-  # TODO: Remove these "shell_version", because they are for service worker functionality we no longer need.
```
Comment (citizen428) [inline app/controllers/async_info_controller.rb:28]: I think we;re safely past mid-March at this point so I removed this here as well as the corresponding route. cc @benhalpern

File: app/controllers/shell_controller.rb [hunk 9]
```diff
@@ -1,19 +0,0 @@
-class ShellController < ApplicationController
-  # TODO: Remove these "ShellController", because they are for service worker functionality we no longer need.
```
Comment (citizen428) [inline app/controllers/shell_controller.rb:2]: I think we're safely past mid-March at this point and I couldn't find any reference to this in the routes. cc @benhalpern

File: app/models/podcast_episode.rb [hunk 10]
```diff
@@ -1,8 +1,4 @@
 class PodcastEpisode < ApplicationRecord
-  self.ignored_columns = %w[
```
Comment (citizen428) [inline app/models/podcast_episode.rb:2]: Verified via Blazer that this is indeed unused and removed the column.

File: app/models/device.rb [hunk 11]
```diff
@@ -1,7 +1,4 @@
 class Device < ApplicationRecord
-  # @fdoxyz to remove app_bundle from Device soon
```
Comment (citizen428) [inline app/models/device.rb:2]: @fdoxyz This comment was from April. I checked with Blazer that we indeed have no data stored in this column and would remove it in this PR if that's ok for you.

**Review by maestromac — APPROVED** [review]
tested, LGTM!

File: app/controllers/async_info_controller.rb (hunk 8 above)
Comment (benhalpern) [reply app/controllers/async_info_controller.rb:28]: Yeah, looks good to me.

**Review by benhalpern — APPROVED** [review]

File: app/models/device.rb (hunk 11 above)
Comment (fdoxyz) [reply app/models/device.rb:2]: Sounds great, thanks for looping back here for all these cleanups! 🙌

**Review by fdoxyz — APPROVED** [review]

**Pull request merged**